Sources/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -509,12 +509,10 @@ extension PurchasesOrchestrator: StoreKitWrapperDelegate {
         guard let delegate = delegate else { return false }
 
         let storeProduct = StoreProduct(sk1Product: product)
-        lock.lock()
         delegate.readyForPromotedProduct(storeProduct) { completion in
-            self.purchaseCompleteCallbacksByProductID[product.productIdentifier] = completion
+            self.purchaseCompleteCallbacksByProductID.modify { $0[product.productIdentifier] = completion }
             storeKitWrapper.add(payment)
```

Comment:
I kinda think it makes sense to add this in to the `.modify` block since that guarantees the same behavior we had with the lock.

Reply:
The thing is this is an `@escaping` block, so the outside `lock` was likely getting unlocked before this was even modified. Does that make sense?

So the access to `purchaseCompleteCallbacksByProductID` wasn't protected. I do see that maybe the intent is that the whole thing was, and therefore maybe `storeKitWrapper.add(payment)` could be inside now. But considering that was missed, and therefore this is the same behavior, maybe this makes more sense?

Reply:
Ahhh, yeah. Makes sense.